src/ppx_tools/traverse_ast2.ml

```diff
@@ -0,0 +1,1936 @@
+open Ppxlib
+
+exception Reparse_error of string
```

Comment:
This exception needs a comment describing the error and the argument.

Reply:
Thanks. This comment is quite helpful. So when this error occurs, we just abort and let it bubble up, right? i.e. there's no way to recover from this error.

PS You might want to review the syntax for [ocamldoc comments](https://ocamlverse.github.io/content/documentation_guidelines.html). It has a different convention for code, lists, etc.

Reply:
On the contrary, we use one argument lift class to continue serializing the subtree that starts where the error occurs. The  subtree will simply not contain double locations (but the ones generated by ppx) which means that simultaneous navigation will not work for that subtree. 

It looks like the following (using `with_reparse` we saw earlier) : 
```OCaml 
method! directive_argument_desc x x' =
      with_reparse
        (fun () -> super#directive_argument_desc x x')
        ~on_reparse_error:(fun () -> parse_ast#directive_argument_desc x)
``` 

I'm reading about odoc, will add comments everywhere soon.

Reply:
I see. In which case, this exception cannot be observed by callers then? Shall we remove it from the mli then?

Reply:
Pretty much, yes.

Reply:
Wait, actually no. It has to be visible at least in the dumpast module, I don't think we can remove it from the mli.

Reply:
In that case, I would suggest the function to return `Result.t`. If the exception needs to be handled, it should just be reflected in the signature.

Reply:
Changed the return type and removed exception fca04409f9d3371d4b799534253d42e1cdbd5631. 
Not sure I can avoid the fact that all `lift2`method return `('res, string) result`. Because this way i have to unwrap some elements with `Result.get_ok` which sould never cause an exception but it's still awkward.